go/vt/mysqlctl/builtinbackupengine.go

```diff
@@ -367,13 +367,19 @@ func (be *BuiltinBackupEngine) ExecuteBackup(ctx context.Context, params BackupP
 		}
 		if !replicationPosition.Equal(masterPos) {
 			for {
-				status, err := mysqld.SlaveStatus()
-				if err != nil {
-					return usable, err
-				}
-				newPos := status.Position
-				if !newPos.Equal(replicationPosition) {
-					break
+				select {
```

Comment:
I don't think you need a `select` to check the context if you're not waiting on anything else at the same time. I usually just do:

```go
if err := ctx.Err(); err != nil {
  return err
}
```